docs/developer/releasing.md

```diff
@@ -0,0 +1,42 @@
+# Releasing
+
+## Output
+
+### Expected artifacts
+
+1. A container image of the shared cluster-api controller manager
+2. A git tag for providers to use
+
+### Artifact locations
+
+1. The container image is found in the registry `gcr.io/k8s-cluster-api` with an
+   image name of `cluster-api-controller` and a tag that matches the release
+   version. For example, in the `0.0.0-alpha.4` release, the container image
+   location is `gcr.io/k8s-cluster-api/cluster-api-controller:0.0.0-alpha.4`
+
+## Process
+
+1. Create a pull request that contains two changes:
```

Comment:
For Velero, we use `goreleaser` and have a custom script that executes at release time (after the git tag has been created) that uses `sed` to find/replace `:latest` with the appropriate version string to match the git tag. That way, we don't have to open a PR, cut the release, then revert the PR. Would you be interested in exploring something like this down the road?

Reply:
I think anything that reduces the number of steps involved with releasing should be considered. I'll add a note in the [Cluster API Provider Releases](https://docs.google.com/document/d/17tY88ysZPDH2rZ9BkewPxETViqivdWDJPrVl_P1TpLg/edit#heading=h.gjrgcawagqn6) doc about goreleaser.